### PR TITLE
fix various compiler warnings

### PR DESCRIPTION
**Pull request by rhaschke**
This PR attempt to fix most compiler warnings in the current melodic-devel branch - there are too many now. I grouped the commits according to types of compiler warnings:
- unused variables / names
- comparison between signed and unsigned
- fix member initialization (typically order)

Fixing those warnings also revealed some errors in the code ;-)

### CODE REVIEW

File: moveit_planners/ompl/ompl_interface/src/constraints_library.cpp [hunk 1]
```diff
@@ -605,4 +605,5 @@ ompl::base::StateStoragePtr ompl_interface::ConstraintsLibrary::constructConstra
 
     return sstor;
   }
+#warning "else case not yet handled: missing return value!"
```
Comment (rhaschke) [inline moveit_planners/ompl/ompl_interface/src/constraints_library.cpp]: @davetcoleman This function is missing a return value. However, I'm not sure what to return here / how to handle the error(?) case. As you are mentioned as a coauthor, you probably know better.

File: moveit_kinematics/srv_kinematics_plugin/src/srv_kinematics_plugin.cpp [hunk 2]
```diff
@@ -328,12 +328,11 @@ bool SrvKinematicsPlugin::searchPositionIK(const std::vector<geometry_msgs::Pose
     if (error_code.val != error_code.SUCCESS)
     {
       ROS_DEBUG_NAMED("srv", "An IK that satisifes the constraints and is collision free could not be found.");
+      // Debug mode for failure:
+      ROS_DEBUG_STREAM("Request was: \n" << ik_srv.request.ik_request);
```
Comment (mlautman) [inline moveit_kinematics/srv_kinematics_plugin/src/srv_kinematics_plugin.cpp]: These  should be ROS_DEBUG_STREAM_NAMED to enable filtering with rosconsole.yaml

File: moveit_planners/chomp/chomp_motion_planner/src/chomp_optimizer.cpp [hunk 3]
```diff
@@ -319,11 +319,11 @@ bool ChompOptimizer::optimize()
 {
   bool optimization_result = 0;
   ros::WallTime start_time = ros::WallTime::now();
-  double averageCostVelocity = 0.0;
-  int currentCostIter = 0;
+  // double averageCostVelocity = 0.0;
```
Comment (mlautman) [inline moveit_planners/chomp/chomp_motion_planner/src/chomp_optimizer.cpp]: Is there a reason to keep these around or can we just delete them?

File: moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_planning.cpp [hunk 4]
```diff
@@ -290,7 +290,7 @@ void MotionPlanningFrame::populatePlannersList(const moveit_msgs::PlannerInterfa
 
   bool found_group = false;
   // the name of a planner is either "GROUP[planner_id]" or "planner_id"
-  if (!group.empty())
+  if (!group.empty()) {
```
Comment (mlautman) [inline moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_planning.cpp]: has this been formatted with clang?

File: moveit_setup_assistant/src/widgets/end_effectors_widget.h [hunk 5]
```diff
@@ -71,7 +71,7 @@ class EndEffectorsWidget : public SetupScreenWidget
 
   EndEffectorsWidget(QWidget* parent, moveit_setup_assistant::MoveItConfigDataPtr config_data);
 
-  /// Recieved when this widget is chosen from the navigation menu
+  /// Received when this widget is chosen from the navigation menu
```
Comment (mlautman) [inline moveit_setup_assistant/src/widgets/end_effectors_widget.h]: remove the third slash

File: moveit_setup_assistant/src/widgets/passive_joints_widget.h [hunk 6]
```diff
@@ -70,7 +70,7 @@ class PassiveJointsWidget : public SetupScreenWidget
 
   PassiveJointsWidget(QWidget* parent, moveit_setup_assistant::MoveItConfigDataPtr config_data);
 
-  /// Recieved when this widget is chosen from the navigation menu
+  /// Received when this widget is chosen from the navigation menu
```
Comment (mlautman) [inline moveit_setup_assistant/src/widgets/passive_joints_widget.h]: remove the third slash

File: moveit_setup_assistant/src/widgets/perception_widget.h [hunk 7]
```diff
@@ -67,10 +67,10 @@ class PerceptionWidget : public SetupScreenWidget
 
   PerceptionWidget(QWidget* parent, moveit_setup_assistant::MoveItConfigDataPtr config_data);
 
-  /// Recieved when this widget is chosen from the navigation menu
+  /// Received when this widget is chosen from the navigation menu
```
Comment (mlautman) [inline moveit_setup_assistant/src/widgets/perception_widget.h]: same

File: moveit_setup_assistant/src/widgets/perception_widget.h [hunk 8]
```diff
@@ -67,10 +67,10 @@ class PerceptionWidget : public SetupScreenWidget
...
   virtual void focusGiven();
 
-  /// Recieved when another widget is chosen from the navigation menu
+  /// Received when another widget is chosen from the navigation menu
```
Comment (mlautman) [inline moveit_setup_assistant/src/widgets/perception_widget.h]: same

File: moveit_setup_assistant/src/widgets/planning_groups_widget.h [hunk 9]
```diff
@@ -86,7 +86,7 @@ class PlanningGroupsWidget : public SetupScreenWidget
 
   void changeScreen(int index);
 
-  /// Recieved when this widget is chosen from the navigation menu
+  /// Received when this widget is chosen from the navigation menu
```
Comment (mlautman) [inline moveit_setup_assistant/src/widgets/planning_groups_widget.h]: same

File: moveit_setup_assistant/src/widgets/robot_poses_widget.h [hunk 10]
```diff
@@ -73,7 +73,7 @@ class RobotPosesWidget : public SetupScreenWidget
 
   RobotPosesWidget(QWidget* parent, moveit_setup_assistant::MoveItConfigDataPtr config_data);
 
-  /// Recieved when this widget is chosen from the navigation menu
+  /// Received when this widget is chosen from the navigation menu
```
Comment (mlautman) [inline moveit_setup_assistant/src/widgets/robot_poses_widget.h]: same

File: moveit_setup_assistant/src/widgets/ros_controllers_widget.h [hunk 11]
```diff
@@ -71,7 +71,7 @@ class ROSControllersWidget : public SetupScreenWidget
 
   void changeScreen(int index);
 
-  /// Recieved when this widget is chosen from the navigation menu
+  /// Received when this widget is chosen from the navigation menu
```
Comment (mlautman) [inline moveit_setup_assistant/src/widgets/ros_controllers_widget.h]: same

File: moveit_setup_assistant/src/widgets/virtual_joints_widget.h [hunk 12]
```diff
@@ -69,7 +69,7 @@ class VirtualJointsWidget : public SetupScreenWidget
 
   VirtualJointsWidget(QWidget* parent, moveit_setup_assistant::MoveItConfigDataPtr config_data);
 
-  /// Recieved when this widget is chosen from the navigation menu
+  /// Received when this widget is chosen from the navigation menu
```
Comment (mlautman) [inline moveit_setup_assistant/src/widgets/virtual_joints_widget.h]: same

File: moveit_kinematics/srv_kinematics_plugin/src/srv_kinematics_plugin.cpp (hunk 2 above)
Comment (rhaschke) [reply moveit_kinematics/srv_kinematics_plugin/src/srv_kinematics_plugin.cpp]: Good point.

File: moveit_planners/chomp/chomp_motion_planner/src/chomp_optimizer.cpp (hunk 3 above)
Comment (rhaschke) [reply moveit_planners/chomp/chomp_motion_planner/src/chomp_optimizer.cpp]: Yes. Down there, is some currently commented code using those variables.

File: moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_planning.cpp (hunk 4 above)
Comment (rhaschke) [reply moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_planning.cpp]: Not yet ;-)

File: moveit_setup_assistant/src/widgets/end_effectors_widget.h (hunk 5 above)
Comment (rhaschke) [reply moveit_setup_assistant/src/widgets/end_effectors_widget.h]: I just automatically replaced the typo. However, I can do the slashes too.

Comment (rhaschke) [reply moveit_setup_assistant/src/widgets/end_effectors_widget.h]: No, stop. Three slashed mark this function to be documented with doxygen.

**Comment by rhaschke** [comment]
@mlautman Addressed your requested changes.

**Comment by mlautman** [comment]
Looks like travis is failing because of changes that were made to moveit_visual_tools. I'm working on a fix over there now.

**Comment by rhaschke** [comment]
@davetcoleman @mlautman Broken Travis is not related to this PR, but to a broken moveit_visual_tools!
https://github.com/ros-planning/moveit_visual_tools/pull/31#issuecomment-415586863

**Comment by rhaschke** [comment]
Sorry, @mlautman. My browser wasn't updated, when I wrote my comment. Thus haven't seen that you already spotted the issue in moveit_visual_tools. Thanks for taking care.

**Comment by v4hn** [comment]
> is junk research code of Ioan's that really should be removed.
> [...] no moveit users actually use it AFAIK

That's not true. We use it in our lab and I pointed it out to others before.
It works nicely, is quite fast, and prevents joint space discontinuities when constraint planning.

We even added a moveit tutorial for it not too long ago.

**Pull request merged**